File: app/src/main/java/com/sakuna63/tumbin/application/di/module/ApplicationModule.java
package com.sakuna63.tumbin.application.di.module;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.realm.RealmConfiguration;

@Module
public class ApplicationModule {
    private final Application app;

    public ApplicationModule(Application app) {
        this.app = app;
    }

    @Provides
    Context context() {
        return app;
    }

    @Provides
    @Singleton
    RealmConfiguration realmConfiguration() {
        return new RealmConfiguration.Builder().build();
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/di/module/ApplicationModule.kt
package com.sakuna63.tumbin.application.di.module

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import io.realm.RealmConfiguration
import javax.inject.Singleton

@Module
open class ApplicationModule(private val app: Application) {

    @Provides
    fun context(): Context = app

    @Provides
    @Singleton
    open fun realmConfiguration(): RealmConfiguration = RealmConfiguration.Builder().build()
}
